### PR TITLE
fix: prevent integer overflow in consensus proof verification

### PR DESCRIPTION
**Pull request by VolodymyrBg**
This commit fixes a critical security vulnerability in the consensus proof
verification system where an attacker could cause integer overflow in signature
size calculations. The vulnerability could potentially lead to memory corruption
and consensus protocol manipulation.

The fix adds proper integer overflow checks using Rust's safe arithmetic
operations (checked_add and checked_mul) when calculating total signature sizes
during proof verification.

Security Impact:
- Prevents potential memory corruption from integer overflow
- Protects against maliciously crafted proofs
- Maintains consensus protocol integrity

Testing:
- Existing tests pass
- Edge cases with large signature counts now safely return None

### CODE REVIEW

**Comment by patrick-ogrady** [comment]
> This commit fixes a critical security vulnerability in the consensus proof
verification system where an attacker could cause integer overflow in signature
size calculations. The vulnerability could potentially lead to memory corruption
and consensus protocol manipulation.

This should only panic on 32-bit architectures but, yes, good find! Would be super annoying when running a wasm-based verifier.

**Comment by patrick-ogrady** [comment]
Would you mind also checking for the same bug in `threshold_simplex`? It shares a good bit of the same code?

File: consensus/src/simplex/prover.rs [hunk 1]
```diff
@@ -150,9 +150,13 @@ impl<C: Scheme, H: Hasher> Prover<C, H> {
         let count = count as usize;
         let message = proposal_message(view, parent, &payload);
 
-        // Decode signatures
+        // Check for integer overflow in size calculation
         let (public_key_len, signature_len) = C::len();
-        if proof.remaining() != count * (public_key_len + signature_len) {
+        let item_size = public_key_len.checked_add(signature_len)?;
```
Comment (patrick-ogrady) [inline consensus/src/simplex/prover.rs:161]: It would be great to write a test that provides a maliciously crafted count.

**Comment by VolodymyrBg** [comment]
@patrick-ogrady  Corrected the code. Thx for quick answer

**Comment by patrick-ogrady** [comment]
If its ok with you, I can take this PR to the finish line from here (some minor nits that'll take me a few moments to sort out)?

**Comment by VolodymyrBg** [comment]
@patrick-ogrady  Of course. I'll be grateful

**Review by patrick-ogrady — APPROVED** [review]
